compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -464,7 +464,26 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                                 root_obligation,
                             )
                         } else {
-                            (trait_predicate, &obligation)
+                            let mut temp_trait_predicate = trait_predicate;
+                            let mut loop_code = obligation.cause.code().clone();
+                            loop{
+                                let (code, base) = loop_code.peel_derives_with_predicate();
```

Comment:
I don't think it's correct to use `peel_derives_with_predicate` here. We need to be peeling these one at a time.

Consider a where clause where the first impl we have to go through isn't marked with `#[do_not_recommend]` but the second is.

Reply:
Hopefully it's more correct with https://github.com/rust-lang/rust/compare/eca373a073222883772da9c30616b3c746e3d6f2..f0279abd052e1ec4fa39ab71edbd7484dbc84c7b now. If that's not the case I would appreciate bit more guidance of how to address that.